controllers/eventsource/controller.go

```diff
@@ -0,0 +1,119 @@
+package eventsource
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+
+	"github.com/argoproj/argo-events/common"
+	"github.com/argoproj/argo-events/pkg/apis/eventsource/v1alpha1"
+)
+
+const (
+	// ControllerName is name of the controller
+	ControllerName = "eventsource-controller"
+
+	finalizerName = ControllerName
+)
+
+type reconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+
+	// controller namespace
+	namespace        string
+	eventSourceImage string
+	logger           logr.Logger
```

Comment:
is there a reason why we are using different log packages like logrus, zap and go-logr?

Reply:
`go-logr` is not a real log package but an interface, the options are just `logrus` and `zap`, I think we should pick `logrus`. The reason here using `zap` is just I didn't get time to figure out how to integrate `logrus` with `go-logr` ( log interface used by `controller-runtime`), but I know it's doable. This will be done later.

Reply:
Or we don't use `go-logr` at all but `logrus` directly in controllers. This shouldn't be a big deal, we can easily change it.

Reply:
As we discussed, will switch to `zap` later.